addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -384,25 +383,46 @@ def _get_accrual_plan_level_work_entry_prorata(self, level, start_period, start_
             work_entry_prorata = worked / (left + planned_worked) if (left + planned_worked) else 0
         return work_entry_prorata
 
-    def _process_accrual_plan_level(self, level, start_period, start_date, end_period, end_date):
-        """
-        Returns the added days for that level
-        """
-        self.ensure_one()
+    def _compute_added_value(self, level, start_period, start_date, end_period, end_date, calendar):
+
         if level.frequency == 'hourly' or level.accrual_plan_id.is_based_on_worked_time:
-            work_entry_prorata = self._get_accrual_plan_level_work_entry_prorata(level, start_period, start_date, end_period, end_date)
+            work_entry_prorata = self._get_accrual_plan_level_work_entry_prorata(level, calendar, start_period, start_date, end_period, end_date)
             added_value = work_entry_prorata * level.added_value
         else:
             added_value = level.added_value
+
         # Convert time in hours to time in days in case the level is encoded in hours
         if level.added_value_type == 'hour':
             added_value = added_value / (self.employee_id.sudo().resource_id.calendar_id.hours_per_day or HOURS_PER_DAY)
+
         period_prorata = 1
         if (start_period != start_date or end_period != end_date) and not level.accrual_plan_id.is_based_on_worked_time:
             period_days = (end_period - start_period)
             call_days = (end_date - start_date)
             period_prorata = min(1, call_days / period_days) if period_days else 1
-        return added_value * period_prorata
+
+        added_value = added_value * period_prorata
+        if level.frequency != 'hourly':
+            added_value *= (calendar.work_time_rate / 100)
+
+        return added_value
+
+    def _process_accrual_plan_level(self, level, start_period, start_date, end_period, end_date):
+        """
+        Returns the added days for that level
+        """
+        self.ensure_one()
+        datetime_min_time = datetime.min.time()
+        employee_timezone = timezone(self.employee_id.tz)
+        start_dt = datetime.combine(start_date, datetime_min_time).replace(tzinfo=employee_timezone)
+        end_dt = datetime.combine(end_date, datetime_min_time).replace(tzinfo=employee_timezone)
+        employee_calendar_periods = self.employee_id._get_calendar_periods(start_dt, end_dt)[self.employee_id]
+        total_added_value = 0
+        for cal_period_start, cal_period_end, calendar in employee_calendar_periods:
+            cal_period_start = cal_period_start.astimezone(employee_timezone)
+            cal_period_end = cal_period_end.astimezone(employee_timezone)
```

Comment:
for example here, for me we should use the timezone defined on the contract :)
Imagine you work from Japan for your first contract and after you moved to Brazil; your second contract will have an other timezone. And the employee timezone will be changed between the two :/